utils/directories.cc

```diff
@@ -145,4 +149,128 @@ future<> directories::verify_owner_and_mode(fs::path path, recursive recursive)
     return do_verify_owner_and_mode(std::move(path), recursive, 0);
 }
 
+static seastar::sstring to_sstring(const fs::path& p) {
+    return seastar::sstring{p.c_str()};
+}
+
+void directories::set_work_dir(const seastar::sstring& work_dir) {
+    _work_dir = fs::path{work_dir};
+}
+
+void directories::set_commitlog_dir(const seastar::sstring& commitlog_dir) {
+    _commitlog_dir = fs::path{commitlog_dir};
+}
+
+void directories::set_schema_commitlog_dir(const seastar::sstring& schema_commitlog_dir) {
+    _schema_commitlog_dir = fs::path{schema_commitlog_dir};
+}
+
+void directories::set_hints_dir(const seastar::sstring& hints_dir) {
+    _hints_dir = fs::path{hints_dir};
+}
+
+void directories::set_view_hints_dir(const seastar::sstring& view_hints_dir) {
+    _view_hints_dir = fs::path{view_hints_dir};
+}
+
+void directories::set_saved_caches_dir(const seastar::sstring& saved_caches_dir) {
+    _saved_caches_dir = fs::path{saved_caches_dir};
+}
+
+void directories::set_data_file_dirs(const std::vector<seastar::sstring>& data_file_dirs) {
+    std::vector<fs::path> dirs_paths;
+    dirs_paths.reserve(data_file_dirs.size());
+
+    for (const auto& dir_str : data_file_dirs) {
+        dirs_paths.emplace_back(dir_str);
+    }
+
+    _data_file_dirs = std::move(dirs_paths);
+}
+
+seastar::sstring directories::get_commitlog_dir() const {
+    return to_sstring(_commitlog_dir);
+}
+
+seastar::sstring directories::get_schema_commitlog_dir() const {
+    return to_sstring(_schema_commitlog_dir);
+}
+
+seastar::sstring directories::get_hints_dir() const {
+    return to_sstring(_hints_dir);
+}
+
+seastar::sstring directories::get_view_hints_dir() const {
+    return to_sstring(_view_hints_dir);
+}
+
+seastar::sstring directories::get_saved_caches_dir() const {
+    return to_sstring(_saved_caches_dir);
+}
+
+std::vector<seastar::sstring> directories::get_data_file_dirs() const {
+    std::vector<seastar::sstring> dirs;
+    dirs.reserve(_data_file_dirs.size());
+
+    for (const auto & dir_path : _data_file_dirs) {
+        dirs.emplace_back(to_sstring(dir_path));
+    }
+
+    return dirs;
+}
+
+void directories::setup_paths(const db::config& cfg) {
+    if (cfg.work_directory.is_set()) {
```

Comment:
I will add a test to verify that this condition does not change the logic that was present before.

`work_directory` may have a default value: `"/var/lib/scylla"`. In `db::config::setup_directories()` the predicate that checks if the variable is set, was not called for this variable.

```
bool is_set() const {
    return _source > config_source::None;
}
```

Reply:
The test showed, that it was an issue. It has been fixed and a new test has been created.